From 53ad152c046a7a4cc535058b329e1339f899f914 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Mon, 11 Nov 2019 17:22:12 +0100
Subject: [PATCH] Remove ObjC

---
 CMakeLists.txt | 43 -------------------------------------------
 1 file changed, 43 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0693fe1f9a..fc1ded0576 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -221,7 +221,6 @@ add_custom_target(plugins
   grpc_cpp_plugin
   grpc_csharp_plugin
   grpc_node_plugin
-  grpc_objective_c_plugin
   grpc_php_plugin
   grpc_python_plugin
   grpc_ruby_plugin
@@ -4743,7 +4742,6 @@ add_library(grpc_plugin_support
   src/compiler/cpp_generator.cc
   src/compiler/csharp_generator.cc
   src/compiler/node_generator.cc
-  src/compiler/objective_c_generator.cc
   src/compiler/php_generator.cc
   src/compiler/python_generator.cc
   src/compiler/ruby_generator.cc
@@ -13624,47 +13622,6 @@ endif()
 endif (gRPC_BUILD_CODEGEN)
 if (gRPC_BUILD_CODEGEN)
 
-add_executable(grpc_objective_c_plugin
-  src/compiler/objective_c_plugin.cc
-)
-
-
-target_include_directories(grpc_objective_c_plugin
-  PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}
-  PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include
-  PRIVATE ${_gRPC_ADDRESS_SORTING_INCLUDE_DIR}
-  PRIVATE ${_gRPC_BENCHMARK_INCLUDE_DIR}
-  PRIVATE ${_gRPC_CARES_INCLUDE_DIR}
-  PRIVATE ${_gRPC_GFLAGS_INCLUDE_DIR}
-  PRIVATE ${_gRPC_PROTOBUF_INCLUDE_DIR}
-  PRIVATE ${_gRPC_SSL_INCLUDE_DIR}
-  PRIVATE ${_gRPC_UPB_GENERATED_DIR}
-  PRIVATE ${_gRPC_UPB_GRPC_GENERATED_DIR}
-  PRIVATE ${_gRPC_UPB_INCLUDE_DIR}
-  PRIVATE ${_gRPC_ZLIB_INCLUDE_DIR}
-  PRIVATE ${_gRPC_PROTO_GENS_DIR}
-)
-
-target_link_libraries(grpc_objective_c_plugin
-  ${_gRPC_PROTOBUF_PROTOC_LIBRARIES}
-  ${_gRPC_PROTOBUF_LIBRARIES}
-  ${_gRPC_ALLTARGETS_LIBRARIES}
-  grpc_plugin_support
-)
-
-
-
-if (gRPC_INSTALL)
-  install(TARGETS grpc_objective_c_plugin EXPORT gRPCTargets
-    RUNTIME DESTINATION ${gRPC_INSTALL_BINDIR}
-    LIBRARY DESTINATION ${gRPC_INSTALL_LIBDIR}
-    ARCHIVE DESTINATION ${gRPC_INSTALL_LIBDIR}
-  )
-endif()
-
-endif (gRPC_BUILD_CODEGEN)
-if (gRPC_BUILD_CODEGEN)
-
 add_executable(grpc_php_plugin
   src/compiler/php_plugin.cc
 )
-- 
2.23.0

