diff --git a/src/core/lib/iomgr/socket_utils_common_posix.cc b/src/core/lib/iomgr/socket_utils_common_posix.cc
index f368cb76f6..aa9037d985 100644
--- a/src/core/lib/iomgr/socket_utils_common_posix.cc
+++ b/src/core/lib/iomgr/socket_utils_common_posix.cc
@@ -30,11 +30,7 @@
 #include <fcntl.h>
 #include <limits.h>
 #include <netinet/in.h>
-#ifdef GRPC_LINUX_TCP_H
 #include <linux/tcp.h>
-#else
-#include <netinet/tcp.h>
-#endif
 #include <stdio.h>
 #include <string.h>
 #include <sys/socket.h>
