From e0e50ee1116d7f1be6d5d34aced013e83ddb6bf3 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Fri, 11 Sep 2020 14:20:04 +0200
Subject: [PATCH 04/15] force protoc executable

---
 cmake/protobuf.cmake | 17 ++---------------
 1 file changed, 2 insertions(+), 15 deletions(-)

diff --git a/cmake/protobuf.cmake b/cmake/protobuf.cmake
index ad80ca6e46..7bae8d95bd 100644
--- a/cmake/protobuf.cmake
+++ b/cmake/protobuf.cmake
@@ -70,21 +70,8 @@ elseif(gRPC_PROTOBUF_PROVIDER STREQUAL "package")
     endif()
     # Well-known proto files are expected to be in the Protobuf include directory.
     get_target_property(_gRPC_PROTOBUF_WELLKNOWN_INCLUDE_DIR ${_gRPC_PROTOBUF_LIBRARIES} INTERFACE_INCLUDE_DIRECTORIES)
-    if(TARGET protobuf::protoc)
-      set(_gRPC_PROTOBUF_PROTOC protobuf::protoc)
-      if(CMAKE_CROSSCOMPILING)
-        find_program(_gRPC_PROTOBUF_PROTOC_EXECUTABLE protoc)
-      else()
-        set(_gRPC_PROTOBUF_PROTOC_EXECUTABLE $<TARGET_FILE:protobuf::protoc>)
-      endif()
-    else()
-      set(_gRPC_PROTOBUF_PROTOC ${PROTOBUF_PROTOC_EXECUTABLE})
-      if(CMAKE_CROSSCOMPILING)
-        find_program(_gRPC_PROTOBUF_PROTOC_EXECUTABLE protoc)
-      else()
-        set(_gRPC_PROTOBUF_PROTOC_EXECUTABLE ${PROTOBUF_PROTOC_EXECUTABLE})
-      endif()
-    endif()
+    set(_gRPC_PROTOBUF_PROTOC_EXECUTABLE ${Protobuf_PROTOC_EXECUTABLE})
+    set(_gRPC_PROTOBUF_PROTOC ${Protobuf_PROTOC_EXECUTABLE})
     set(_gRPC_FIND_PROTOBUF "include(CMakeFindDependencyMacro)\nfind_dependency(Protobuf CONFIG)")
   endif()
 endif()
