From 795a8dea3cba2958b01131bc9c9c68288bf7a5ce Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Wed, 31 Aug 2022 18:52:58 +0200
Subject: [PATCH 5/9] patch tests to use passed CXX_STANDARD

---
 examples/cpp/cmake/common.cmake | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/examples/cpp/cmake/common.cmake b/examples/cpp/cmake/common.cmake
index 3e12e2d6d4..1326a5be3a 100644
--- a/examples/cpp/cmake/common.cmake
+++ b/examples/cpp/cmake/common.cmake
@@ -19,7 +19,9 @@
 
 cmake_minimum_required(VERSION 3.5.1)
 
-set (CMAKE_CXX_STANDARD 14)
+if (NOT DEFINED CMAKE_CXX_STANDARD)
+  set (CMAKE_CXX_STANDARD 14)
+endif()
 
 if(MSVC)
   add_definitions(-D_WIN32_WINNT=0x600)
-- 
2.37.0.windows.1

