From 3fe072b196fec268a6d1e4f1ae3bd59efd0d4bb3 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 4 Sep 2022 14:34:52 +0200
Subject: [PATCH 6/6] fix abseil setup

---
 setup.py | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/setup.py b/setup.py
index 0e535e259a..fee180d4ef 100644
--- a/setup.py
+++ b/setup.py
@@ -340,13 +340,16 @@ if BUILD_WITH_SYSTEM_CARES:
 if BUILD_WITH_SYSTEM_RE2:
     EXTENSION_LIBRARIES += ('re2',)
 if BUILD_WITH_SYSTEM_ABSL:
-    EXTENSION_LIBRARIES += tuple(
-        lib.stem[3:] for lib in pathlib.Path('/usr').glob('lib*/libabsl_*.so'))
+    if "win32" in sys.platform:
+        absl_libs = ('abseil_dll', 'absl_flags',)
+    else:
+        absl_glob = pathlib.Path(os.environ['PREFIX']).glob('lib/libabsl_*.so')
+        absl_libs = tuple(lib.stem[3:] for lib in absl_glob)
+    EXTENSION_LIBRARIES += absl_libs
 if BUILD_WITH_SYSTEM_GRPC_CORE:
     EXTENSION_LIBRARIES += ('gpr', 'grpc', )
     if "win32" in sys.platform:
         EXTENSION_LIBRARIES += ('libprotoc', 'libprotobuf', 'libprotobuf-lite', 'upb', 'address_sorting', )
-        EXTENSION_LIBRARIES += tuple(lib.stem for lib in pathlib.Path(os.environ['LIBRARY_LIB']).glob('absl_*.lib'))
 
 DEFINE_MACROS = (('_WIN32_WINNT', 0x600),)
 asm_files = []
-- 
2.38.1.windows.1

