From f7dc32ad90c8ecfdab093a38db55123610a07570 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 9 Jun 2023 22:16:09 +1100
Subject: [PATCH 09/10] don't use find_dependency for protobuf

---
 cmake/protobuf.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/protobuf.cmake b/cmake/protobuf.cmake
index 96213e7523..28f3498b78 100644
--- a/cmake/protobuf.cmake
+++ b/cmake/protobuf.cmake
@@ -73,6 +73,6 @@ elseif(gRPC_PROTOBUF_PROVIDER STREQUAL "package")
     endif()
     set(_gRPC_PROTOBUF_PROTOC_EXECUTABLE ${Protobuf_PROTOC_EXECUTABLE})
     set(_gRPC_PROTOBUF_PROTOC ${Protobuf_PROTOC_EXECUTABLE})
-    set(_gRPC_FIND_PROTOBUF "find_dependency(Protobuf CONFIG)")
+    set(_gRPC_FIND_PROTOBUF "find_package(Protobuf CONFIG)")
   endif()
 endif()
