From 897c1489853f71d27cbfd549de11ab8fcfcd449d Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 1 Mar 2025 08:06:28 +1100
Subject: [PATCH 14/18] add linkage to openssl to grpc_unsecure

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0a1b20ada5..7faf19eebf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3867,6 +3867,7 @@ target_link_libraries(grpc_unsecure
   PRIVATE ${_gRPC_ZLIB_LIBRARIES}
   PRIVATE ${_gRPC_CARES_LIBRARIES}
   # external dependencies that need to available at runtime
+  PRIVATE ${_gRPC_SSL_LIBRARIES}
   PRIVATE absl::algorithm_container
   PRIVATE absl::config
   PRIVATE absl::no_destructor
