From 44017807ebc0c0a3af9ae6957c1507f0e7146a45 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 3 Mar 2025 14:05:51 +1100
Subject: [PATCH 15/22] make stuff private

---
 CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 67e4b0b500..9fc379d81c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3075,9 +3075,9 @@ target_include_directories(grpc
 )
 target_link_libraries(grpc
   # core libs
-  PUBLIC  ${_gRPC_ALLTARGETS_LIBRARIES}
+  PRIVATE ${_gRPC_ALLTARGETS_LIBRARIES}
   # vendored libs
-  PUBLIC  ${_gRPC_ADDRESS_SORTING_LIBRARIES}
+  PRIVATE ${_gRPC_ADDRESS_SORTING_LIBRARIES}
   # external dependencies used within grpc
   PRIVATE ${_gRPC_CARES_LIBRARIES}
   PRIVATE ${_gRPC_RE2_LIBRARIES}
@@ -3773,9 +3773,9 @@ target_include_directories(grpc_unsecure
 )
 target_link_libraries(grpc_unsecure
   # core libs
-  PUBLIC  ${_gRPC_ALLTARGETS_LIBRARIES}
+  PRIVATE ${_gRPC_ALLTARGETS_LIBRARIES}
   # vendored libs
-  PUBLIC  ${_gRPC_ADDRESS_SORTING_LIBRARIES}
+  PRIVATE ${_gRPC_ADDRESS_SORTING_LIBRARIES}
   # external dependencies used within grpc
   PRIVATE ${_gRPC_ZLIB_LIBRARIES}
   PRIVATE ${_gRPC_CARES_LIBRARIES}
