From ef2bf624be9f9de8bcdaf1c038b6113fc22b5a11 Mon Sep 17 00:00:00 2001
From: Tomasz Kaczkowski <tomasz.kaczkowski@volue.com>
Date: Wed, 16 Apr 2025 12:40:50 +0200
Subject: [PATCH 16/16] Add macro protection files to cc files

---
 src/compiler/cpp_generator.cc | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/compiler/cpp_generator.cc b/src/compiler/cpp_generator.cc
index 7347960fe1..5b795a6f01 100644
--- a/src/compiler/cpp_generator.cc
+++ b/src/compiler/cpp_generator.cc
@@ -1757,7 +1757,9 @@ std::string GetSourceIncludes(grpc_generator::File* file,
         "grpcpp/impl/server_callback_handlers.h",
         "grpcpp/server_context.h",
         "grpcpp/impl/service_type.h",
-        "grpcpp/support/sync_stream.h"};
+        "grpcpp/support/sync_stream.h",
+        // ports_def.inc Must be included as last
+        "grpcpp/ports_def.inc"};
     std::vector<std::string> headers(headers_strs, array_end(headers_strs));
     PrintIncludes(printer.get(), headers, params.use_system_headers,
                   params.grpc_search_path);
@@ -2253,6 +2255,8 @@ std::string GetSourceEpilogue(grpc_generator::File* file,
       temp.append(*part);
       temp.append("\n");
     }
+    // Must be included at end of file
+    temp.append("#include <grpcpp/ports_undef.inc>\n");
     temp.append("\n");
   }
 
