From 2779912da9758059f41012f13468ca44932d70bd Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Tue, 26 Sep 2023 18:25:54 +1100
Subject: [PATCH 16/16] add missing linkages for gRPC_BUILD_TESTS=ON

---
 CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 836fb1daa7..443ea381c6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6328,6 +6328,7 @@ target_include_directories(authorization_matchers_test
 target_link_libraries(authorization_matchers_test
   ${_gRPC_BASELIB_LIBRARIES}
   ${_gRPC_PROTOBUF_LIBRARIES}
+  ${_gRPC_RE2_LIBRARIES}
   ${_gRPC_ZLIB_LIBRARIES}
   ${_gRPC_ALLTARGETS_LIBRARIES}
   grpc_test_util
@@ -13061,6 +13062,7 @@ target_include_directories(grpc_authorization_engine_test
 target_link_libraries(grpc_authorization_engine_test
   ${_gRPC_BASELIB_LIBRARIES}
   ${_gRPC_PROTOBUF_LIBRARIES}
+  ${_gRPC_RE2_LIBRARIES}
   ${_gRPC_ZLIB_LIBRARIES}
   ${_gRPC_ALLTARGETS_LIBRARIES}
   grpc_test_util
@@ -14890,6 +14892,7 @@ if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   target_link_libraries(httpcli_test
     ${_gRPC_BASELIB_LIBRARIES}
+    ${_gRPC_CARES_LIBRARIES}
     ${_gRPC_PROTOBUF_LIBRARIES}
     ${_gRPC_ZLIB_LIBRARIES}
     ${_gRPC_ALLTARGETS_LIBRARIES}
@@ -16392,6 +16395,7 @@ target_include_directories(matchers_test
 target_link_libraries(matchers_test
   ${_gRPC_BASELIB_LIBRARIES}
   ${_gRPC_PROTOBUF_LIBRARIES}
+  ${_gRPC_RE2_LIBRARIES}
   ${_gRPC_ZLIB_LIBRARIES}
   ${_gRPC_ALLTARGETS_LIBRARIES}
   grpc_test_util
@@ -19789,6 +19793,7 @@ target_include_directories(resolve_address_using_ares_resolver_test
 
 target_link_libraries(resolve_address_using_ares_resolver_test
   ${_gRPC_BASELIB_LIBRARIES}
+  ${_gRPC_CARES_LIBRARIES}
   ${_gRPC_PROTOBUF_LIBRARIES}
   ${_gRPC_ZLIB_LIBRARIES}
   ${_gRPC_ALLTARGETS_LIBRARIES}
@@ -19892,6 +19897,7 @@ target_include_directories(resolve_address_using_native_resolver_test
 
 target_link_libraries(resolve_address_using_native_resolver_test
   ${_gRPC_BASELIB_LIBRARIES}
+  ${_gRPC_CARES_LIBRARIES}
   ${_gRPC_PROTOBUF_LIBRARIES}
   ${_gRPC_ZLIB_LIBRARIES}
   ${_gRPC_ALLTARGETS_LIBRARIES}
