From f6dfc9e7010f8aefdae71b9e4d11d4ef2ca93af9 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Tue, 26 Sep 2023 18:25:54 +1100
Subject: [PATCH 16/16] add missing linkages for tests

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 836fb1daa7..0588edf039 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6329,6 +6329,7 @@ target_link_libraries(authorization_matchers_test
   ${_gRPC_BASELIB_LIBRARIES}
   ${_gRPC_PROTOBUF_LIBRARIES}
   ${_gRPC_ZLIB_LIBRARIES}
+  ${_gRPC_RE2_LIBRARIES}
   ${_gRPC_ALLTARGETS_LIBRARIES}
   grpc_test_util
 )
