From 5ee7bce14964f0e44d2d82b0ab7f07a76bccd7c0 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 28 Feb 2025 19:42:43 +1100
Subject: [PATCH 18/20] build descriptor.upb_minitable.c as part of
 upb_json_lib

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a98c4d1d40..1ddebab2fc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4138,6 +4138,7 @@ add_library(upb_json_lib ${_gRPC_STATIC_WIN32}
   third_party/upb/upb/reflection/method_def.c
   third_party/upb/upb/reflection/oneof_def.c
   third_party/upb/upb/reflection/service_def.c
+  src/core/ext/upb-gen/google/protobuf/descriptor.upb_minitable.c
 )
 
 target_compile_features(upb_json_lib PUBLIC cxx_std_14)
