From e36f5a767d1d09ca049562d963e9099c8e41bbd0 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 1 Mar 2025 08:06:28 +1100
Subject: [PATCH 19/22] add linkage to openssl to grpc_unsecure

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ebbd283ca5..86238a3e6a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3866,6 +3866,7 @@ target_link_libraries(grpc_unsecure
   PRIVATE ${_gRPC_ZLIB_LIBRARIES}
   PRIVATE ${_gRPC_CARES_LIBRARIES}
   # external dependencies that need to available at runtime
+  PUBLIC  ${_gRPC_SSL_LIBRARIES}
   PUBLIC  absl::algorithm_container
   PUBLIC  absl::config
   PUBLIC  absl::no_destructor
