From 573d0b18cb7b90fd3ac231f6e917c2476ba0cadc Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 2 Mar 2025 09:59:22 +1100
Subject: [PATCH 21/22] point to external upb

---
 CMakeLists.txt  | 2 ++
 cmake/upb.cmake | 4 +---
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ad9abac278..976576e30e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4257,6 +4257,7 @@ target_include_directories(grpc++
 )
 target_link_libraries(grpc++
   ${_gRPC_ALLTARGETS_LIBRARIES}
+  protobuf::libupb
   absl::absl_check
   absl::absl_log
   grpc
@@ -5000,6 +5001,7 @@ target_include_directories(grpc++_unsecure
 )
 target_link_libraries(grpc++_unsecure
   ${_gRPC_ALLTARGETS_LIBRARIES}
+  protobuf::libupb
   absl::absl_check
   absl::absl_log
   grpc_unsecure
diff --git a/cmake/upb.cmake b/cmake/upb.cmake
index 9156e5f48f..e6fa81932d 100644
--- a/cmake/upb.cmake
+++ b/cmake/upb.cmake
@@ -12,9 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set(UPB_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/third_party/upb)
-
-set(_gRPC_UPB_INCLUDE_DIR "${UPB_ROOT_DIR}" "${CMAKE_CURRENT_SOURCE_DIR}/third_party/utf8_range")
+set(_gRPC_UPB_INCLUDE_DIR "${CMAKE_PREFIX_PATH}/include")
 set(_gRPC_UPB_GRPC_GENERATED_DIR "${CMAKE_CURRENT_SOURCE_DIR}/src/core/ext/upb-gen" "${CMAKE_CURRENT_SOURCE_DIR}/src/core/ext/upbdefs-gen")
 
 set(_gRPC_UPB_LIBRARIES upb)
